### ISSUE TITLE
Spock-Reports doesn't support Spock 1.0

### ISSUE THREAD

**Issue by mzielinski** [renatoathaydes/spock-reports#33]
It works with Spock v0.7-groovy-2.0, but it doesn't work with Spock v1.0-groovy-2.0.
The root cause is that the IGlobalExtension interface changed and it now has start() and stop() methods, which the v0.7 version did not have, and so an AbstractMethodError: null is thrown:

{code}
java.lang.AbstractMethodError: null
        at org.spockframework.runtime.GlobalExtensionRegistry.startGlobalExtensions(GlobalExtensionRegistry.java:105)
        at org.spockframework.runtime.RunContext.start(RunContext.java:63)
        at org.spockframework.runtime.RunContext.get(RunContext.java:165)
        at org.spockframework.runtime.Sputnik.runExtensionsIfNecessary(Sputnik.java:88)
        at org.spockframework.runtime.Sputnik.run(Sputnik.java:61)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
{code}

Because of this, we cannot use the most recent features in Spock, since v0.7 is more than 2 years old.

**Comment by renatoathaydes** [comment]
Ok, I can fix that, but 1.0 being a SNAPSHOT it is possible it will break again soon after.... anyway, let's try to fix this, then hope it will stay stable until 1.0 is finally released!

**Comment by renatoathaydes** [comment]
Oh, wow, completely new to me!!! 1.0 was released!!! Wow, finally :) Great, I will make this a priority!

Thanks for raising the issue.

**Comment by mzielinski** [comment]
No problem. Thanks. I was also positively surprised that we have finally spock 1.0 released version :) By mistake I add a word "SNAPHOST" in pull request commit comment. Of corse I had in mind released version.

**Comment by renatoathaydes** [comment]
After upgrading to Spock 1.0, I have been able to compile and run Specs using Spock version `1.0-groovy-2.0` on both Java 7 and Java 8.

The other versions don't work with neither Java 7 nor Java 8! Nothing to do with spock-reports, it is Spock that seems to break.

I tried using Java 7 update 75 and Java 8 update 31.

When using the following combinations:

* java7_u75 `groovy-all:2.3.10` and `spock-core:1.0-groovy-2.3`
* java8_u31 `groovy-all:2.3.10` and `spock-core:1.0-groovy-2.3`
* java7_u75 `groovy-all:2.4.1` and `spock-core:1.0-groovy-2.4`
* java8_u31 `groovy-all:2.4.1` and `spock-core:1.0-groovy-2.4`

I always get this error:

```
java.lang.UnsupportedClassVersionError: test/SimpleOne : Unsupported major.minor version 52.0
```

I will post this in the Spock user group and see if they know what's going on... my guess is that they compiled the spock jars with Java 7 and 8 even newer than mine! Which is a little extreme, they should probably have used some older updates.

I cannot release support for Spock 1.0 until this is resolved.

**Comment by renatoathaydes** [comment]
Turns out the above problem was just due to my test project using `sourceCompatibility = 1.8`.

After changing to 1.7, it just worked!

So I will release spock-reports in a few minutes... the same version works for any combination of java/groovy/spock, which is great!

**Comment by renatoathaydes** [comment]
Spock-reports version 1.2.5 released with support for both Spock 0.7 and all 1.0-xxx versions!
